miden/src/repl/mod.rs

```diff
@@ -270,18 +269,15 @@ fn execute(program: String) -> Result<(Vec<(u64, Word)>, Vec<Felt>), ProgramErro
     let stack_inputs = StackInputs::default();
     let advice_provider = MemAdviceProvider::default();
 
-    let mut process = Process::new_debug(program.kernel().clone(), stack_inputs, advice_provider);
-    let _program_outputs = process.execute(&program).map_err(ProgramError::ExecutionError);
-
-    let (sys, _, stack, _, chiplets, _) = process.into_parts();
+    let state_iter = processor::execute_iter(&program, stack_inputs, advice_provider);
+    let (system, _, stack, chiplets) = state_iter.into_parts();
```

Comment:
I like this approach but I think one thing that's missing is that it won't return an error in case of failed execution. We should probably include the error object in the `into_parts()` method, and if it is not `None` return an error from here as well.

Reply:
Yes, that makes sense. I'll send a commit.

Reply:
Addressed in 381ec959